glib/src/gstring.rs

```diff
@@ -511,6 +579,32 @@ mod tests {
         assert_eq!(s.as_str(), "foo");
     }
 
+    #[test]
+    fn test_from_glib_container() {
+        unsafe {
+            let a: GString = FromGlibContainer::from_glib_container_num(
+                ffi::g_strdup("hello_world".as_ptr() as *const i8),
+                5,
+            );
+            assert_eq!("hello", a.as_str());
+            let b: GString = FromGlibContainer::from_glib_none_num("hello_world".as_ptr(), 5);
+            assert_eq!("hello", b.as_str());
+
+            let c: GString =
+                FromGlibContainer::from_glib_none_num(std::ptr::null() as *const i8, 0);
+            assert_eq!("", c.as_str());
+
+            let d: GString = FromGlibContainer::from_glib_none_num("".as_ptr(), 0);
+            assert_eq!("", d.as_str());
+
+            let e: GString = FromGlibContainer::from_glib_container_num(
+                ffi::g_strdup(std::ptr::null() as *const i8),
+                0,
```

Comment:
And I think this actually is UB because it calls `slice::from_raw_parts()` somewhere with `NULL`. See https://doc.rust-lang.org/std/slice/fn.from_raw_parts.html

The `as_str()` and other functions further above would need to special-case `NULL` / `0` and simply return a `""` in that case.

Reply:
Does that has to happen on all other functions? otherwise I think we can just add a check to `full` & `container` implementation, the same way I did for the `none` one

Reply:
Well, `GString::from_glib_full(ptr::null())` seems to be allowed and would end up in exactly the same situation (before your changes already).

Probably best to handle that generally somewhere.

Reply:
It should be handled now